app/vue/src/client/preview/render.js

```diff
@@ -1,12 +1,49 @@
 import { stripIndents } from 'common-tags';
 import Vue from 'vue';
 
-let app = null;
+let root = null;
 
-function renderRoot(options) {
-  if (app) app.$destroy();
+function updateComponent(component) {
+  const [lastStory] = root.$children;
 
-  app = new Vue(options);
+  if (!lastStory) {
+    return false;
+  }
+
+  if (typeof component.data !== 'function') {
+    return false;
+  }
+
+  const data = component.data();
+
+  if (!data) {
+    return false;
+  }
+
+  Object.entries(data).forEach(([key, value]) => {
+    lastStory[key] = value;
+  });
+
+  lastStory.$forceUpdate();
```

Comment:
I don't believe this will be necessary since Vue as observers for all values of component internal states ; but that won't kill performances either for most of the case, so it's ok to leave it if you feel it's safer.